python/src/scipp/plot/controller.py

```diff
@@ -284,17 +284,32 @@ def rescale_to_data(self, button=None):
                 vmin = self.vmin
             else:
                 self.vmin = vmin
+        else:
+            self.vmin = vmin
         if self.vmax is not None:
             if button is None:
                 vmax = self.vmax
             else:
                 self.vmax = vmax
+        else:
+            self.vmax = vmax
         vmin, vmax = check_log_limits(vmin=vmin, vmax=vmax, scale=self.norm)
         self.view.rescale_to_data(vmin, vmax)
         if self.panel is not None:
             self.panel.rescale_to_data(vmin=vmin,
                                        vmax=vmax,
                                        mask_info=self.get_masks_info())
+        self.refresh()
+
+    def refresh(self):
+        """
+        Get current slice values from model and update the view.
+        This is used when either vmin and vmax limits have been changed, or
+        the normalization has been modified.
+        """
+        new_values = self.model.get_slice_values(
```

Comment:
This would involve caching the values in the `controller`, instead of keep the data in the `model`.
The data values are already cached in the `model` (in the `self.dslice` variable), meaning that all the slicing and resmapling has been done.
We could also cache the masks brodcasting and the transposing etc but I think it's ok for now.